src/main/java/com/iexec/core/replicate/ReplicatesService.java

```diff
@@ -328,6 +334,25 @@ public void updateReplicateStatus(String chainTaskId,
         return Optional.ofNullable(nextAction); // should we return a default action when null?
     }
 
+    @Recover
+    public void updateReplicateStatus(OptimisticLockingFailureException exception,
+                                      String chainTaskId,
+                                      String walletAddress,
+                                      ReplicateStatus newStatus) {
+        log.error("Could not update replicate status, maximum number of retries reached");
+        exception.printStackTrace();
+    }
+
+    @Recover
+    public void updateReplicateStatus(OptimisticLockingFailureException exception,
+                                      String chainTaskId,
+                                      String walletAddress,
+                                      ReplicateStatus newStatus,
+                                      ReplicateStatusDetails details) {
+        log.error("Could not update replicate status, maximum number of retries reached");
+        exception.printStackTrace();
+    }
```

Comment:
I think it's better to have each recovery method directly after its own retryable method.

Reply:
Sure, thanks!